### PR TITLE
refactor: Refactored MASTER logits decoding

### PR DESCRIPTION
**Pull request by fg-mindee**
This PR introduces the following changes:
- removes the decoding buffer from the returned values of `self.decode` for MASTER
- cleans up the logits computation

Any feedback is welcome!

### CODE REVIEW

**Review by charlesmindee — DISMISSED** [review]
Thanks for the refacto!

**Review by charlesmindee — APPROVED** [review]
Thanks!
